src/gamma_inc.jl

```diff
@@ -421,39 +421,30 @@ See also: [`gamma_inc(a,x,ind)`](@ref SpecialFunctions.gamma_inc)
 """
 function gamma_inc_taylor(a::Float64, x::Float64, ind::Integer)
     acc = acc0[ind + 1]
-    wk = zeros(30)
-    flag = false
-    apn = a + 1.0
-    t = x/apn
-    wk[1] = t
-    loop = 2
-    for indx = 2:20
-       apn += 1.0
-       t *= x/apn
-       if t <= 1.0e-3
-           loop = indx
-           flag = true
-           break
-       end
-       wk[indx] = t
-    end
-    if !flag
-        loop = 20
-    end
-    sm = t
-    tol = 0.5*acc #tolerance
-    while true
-       apn += 1.0
-       t *= x/apn
-       sm += t
-       if t <= tol
-           break
-       end
+
+    # compute first 21 terms
```

Comment:
In theory better for numerical stability and precision to do it this way, because otherwise you can lose the accumulated effect of the small components when they're very much smaller than the large components, but I don't know in practice if/when it affects results of this function.

Reply:
> does anyone know where the current implementation was copied from?

ACM TOMS (FREE ACCESS): https://dl.acm.org/doi/abs/10.1145/22721.23109
Fig. 2 in the paper shows the flowchart of the algorithm as a whole.

- `gamma_inc_taylor`: Equ. 15
- `gamma_inc_asym`: Equ. 16

The original paper is mentioned in the code comments.
Perhaps those comments should be moved to the function documentation.

https://github.com/JuliaMath/SpecialFunctions.jl/blob/fdf95ab0865fcd8f3156e118fe5a5c6fd3bc5b66/src/gamma_inc.jl#L831-L834

Reply:
In that case maybe this is the FORTRAN code itself? https://jblevins.org/mirror/amiller/inc_gam.f90